### PR TITLE
Created Search view

### PR DESCRIPTION
**Pull request by ConnorFoody**
#15

### CODE REVIEW

File: MeNext/Layout/SearchView.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	{
+		private SearchBar searchBar;
+		private Label resultsLabel;
+		private SongListView songList;
+
+		public SearchView()
+		{
+			// TODO: clean up this testing code
+			this.Title = "Search";
+			NavigationPage.SetHasNavigationBar(this, false);
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:19]: Why not use something like a horizontal layout here?

File: MeNext/Layout/SearchView.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	public class SearchView : ContentPage
+	{
+		private SearchBar searchBar;
+		private Label resultsLabel;
+		private SongListView songList;
+
+		public SearchView()
+		{
+			// TODO: clean up this testing code
+			this.Title = "Search";
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:18]: Isn't title given by the main page?

File: MeNext/Layout/SearchView.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			NavigationPage.SetHasNavigationBar(this, false);
+			Song selectedSong = null;
+
+			resultsLabel = new Label
+			{
+				Text = "Results will appear below.",
+				//VerticalOptions = LayoutOptions.FillAndExpand,
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:25]: Remove commented out lines

File: MeNext/Layout/SearchView.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+				Orientation = StackOrientation.Horizontal,
+				VerticalOptions = LayoutOptions.FillAndExpand,
+				Children = {
+					playNextButton,
+					suggestionButton
+				}
+			};
+
+			//var model = new SongListModel(new List<Song>());
+			songList = new SongListView(new SongListModel(new List<Song>()), new BasicSongCellFactory());
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:55]: We need access to the model to update it...

File: MeNext/Layout/SearchView.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+			{
+				searchResults = songs2;
+			}
+			else
+			{
+				searchResults = songs;
+			}
+
+			//TODO: make this work with the SongListView better; use the Songs variable, or something similar?
+			songList.ItemsSource = searchResults;
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:117]: we should update the existing model, not replace...

File: MeNext/Layout/SearchView.cs [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+				}
+			};
+		}
+
+		public void TextChanged(string text)
+		{
+			//TODO: get songs from music service
+			resultsLabel.Text = "Searching for: " + text;
+
+			List<Song> songs = new List<Song> { new Song("No"), new Song("Results"), };
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:86]: We should set the "no results" on the label or something

File: MeNext/Layout/SearchView.cs [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+					resultsLabel,
+					songList,
+				}
+			};
+		}
+
+		public void TextChanged(string text)
+		{
+			//TODO: get songs from music service
+			resultsLabel.Text = "Searching for: " + text;
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:84]: User already knows what they are searching for, is this just a testing thing?

Comment (ConnorFoody) [reply MeNext/Layout/SearchView.cs:84]: maybe hide it when searching?

File: MeNext/Layout/SearchView.cs [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+			songList = new SongListView(new SongListModel(new List<Song>()), new BasicSongCellFactory());
+			songList.OnSongSelected += (song) =>
+			{
+				Debug.WriteLine("selected song: " + song.Name);
+				selectedSong = song;
+			};
+
+			searchBar = new SearchBar
+			{
+				Placeholder = "Enter search term",
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:64]: I don't know that we need text here, we can get into this later though

File: MeNext/Layout/SearchView.cs [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+		public SearchView()
+		{
+			// TODO: clean up this testing code
+			this.Title = "Search";
+			NavigationPage.SetHasNavigationBar(this, false);
+			Song selectedSong = null;
+
+			resultsLabel = new Label
+			{
+				Text = "Results will appear below.",
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:24]: I feel like the user can guess this from the visual queues provided / will figure it out once they start searching

File: MeNext/Layout/SearchView.cs [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+				searchResults = songs2;
+			}
+			else
+			{
+				searchResults = songs;
+			}
+
+			//TODO: make this work with the SongListView better; use the Songs variable, or something similar?
+			songList.ItemsSource = searchResults;
+			//songList.Songs = new SongListModel(searchResults);
```
Comment (ConnorFoody) [inline MeNext/Layout/SearchView.cs:118]: remove this line?

**Review by ConnorFoody — COMMENTED** [review]
Looks good!

Some little UI nits in the comments. 

Biggest issue is use of the model. It is observed, so any changes you make to it will automagically be picked up by the UI. We should stick to interacting with it directly instead of with the UI...

Also maybe add some docs explaining whats going on. I know the code is pretty straight forwards, but just a few lines for the teacher would be good

**Comment by daniel-centore** [comment]
Sorry, I should not have merged without checking that it compiled. I am still having the issue:

`/Users/daniel/Projects/menext-app/iOS/Assets.xcassets/AppIcon.appiconset/MeNextLogo1-87.png: error : File not found: /Users/daniel/Projects/menext-app/iOS/Assets.xcassets/AppIcon.appiconset/MeNextLogo1-87.png`

File: MeNext/Layout/SearchView.cs (hunk 7 above)
Comment (samsaksfithian) [reply MeNext/Layout/SearchView.cs:84]: Yes, it was just a testing thing, I should have made that more clear.

File: MeNext/Layout/SearchView.cs (hunk 1 above)
Comment (samsaksfithian) [reply MeNext/Layout/SearchView.cs:19]: By default these types of pages do have a Navigation Bar thing at the top, but only the overall page actually has/needs one, this line just makes it so there isn't a weird empty line at the top of the page.

**Pull request merged**